### PR TITLE
[IMP] contract: Automatic prices for lines

### PR DESCRIPTION
**Pull request by pedrobaeza**
Includes migration script for backwards compatibility

@Tecnativa

### CODE REVIEW

File: contract/models/account_analytic_invoice_line.py [hunk 1]
```diff
@@ -61,6 +71,35 @@ class AccountAnalyticInvoiceLine(models.Model):
...
+        """Get the specific price if no auto-price, and the price obtained
+        from the pricelist otherwise.
+        """
+        for line in self:
+            if line.automatic_price:
+                product = line.product_id.with_context(
+                    quantity=line.quantity,
+                    pricelist=line.analytic_account_id.pricelist_id.id,
+                    partner=line.analytic_account_id.partner_id.id,
+                    date=line.env.context.get('old_date', fields.Date.today()),
```
Comment (carlosdauden) [inline contract/models/account_analytic_invoice_line.py:92]: All OK when create invoice, but recurring_next_date will be ignored on view.

**Review by carlosdauden — APPROVED** [review]
Minor comment

File: contract/models/account_analytic_invoice_line.py (hunk 1 above)
Comment (pedrobaeza) [reply contract/models/account_analytic_invoice_line.py:92]: Well, when adding lines, we will see the price for the current date, which is more than acceptable IMO.

**Review by lasley — APPROVED** [review]
Thanks @pedrobaeza

**Pull request merged**